src/tms_service.ts

```diff
@@ -193,21 +193,37 @@ export class TMSService extends AbstractEmsService {
     return this._emsClient.getValueInLanguage(this._config.name);
   }
 
-  async getMinZoom(): Promise<number | undefined> {
-    const tileJson = await this._getRasterStyleJson();
-    if (tileJson) {
-      return tileJson.minzoom;
-    } else {
-      return;
+  async getMinZoom(format = 'vector'): Promise<number | undefined> {
+    switch (format) {
+      case 'vector':
+        const { sources } = (await this._getVectorStyleJsonInlined()) || { sources: {} };
+        return Math.min(
+          ...Object.values(sources)
+            .map(({ minzoom }) => minzoom)
+            .filter((minzoom): minzoom is number => Number.isFinite(minzoom))
```

Comment:
nit: I'd keep this more simple since the team owns the full stack here (data-generation, EMS, client-library, Map and Vega clients). We know what it should be.

`return sources[0].minzoom`.